crates/solidity/outputs/cargo/crate/src/backend/ast/generated/visitor.rs

```diff
@@ -0,0 +1,2701 @@
+
+pub fn accept_using_alias(target: &UsingAlias, visitor: &mut dyn Visitor) {
+    if visitor.enter_using_alias(target) {
+        accept_using_operator(&target.operator, visitor);
+    }
+    visitor.leave_using_alias(target);
+}
+
+pub fn accept_contract_definition(target: &ContractDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_contract_definition(target) {
+        accept_contract_specifiers(&target.specifiers, visitor);
+        accept_contract_members(&target.members, visitor);
+    }
+    visitor.leave_contract_definition(target);
+}
+
+pub fn accept_inheritance_specifier(target: &InheritanceSpecifier, visitor: &mut dyn Visitor) {
+    if visitor.enter_inheritance_specifier(target) {
+        accept_inheritance_types(&target.types, visitor);
+    }
+    visitor.leave_inheritance_specifier(target);
+}
+
+pub fn accept_inheritance_type(target: &InheritanceType, visitor: &mut dyn Visitor) {
+    if visitor.enter_inheritance_type(target) {
+        accept_identifier_path(&target.type_name, visitor);
+        if let Some(ref arguments) = target.arguments {
+            accept_arguments_declaration(arguments, visitor);
+        }
+    }
+    visitor.leave_inheritance_type(target);
+}
+
+pub fn accept_storage_layout_specifier(target: &StorageLayoutSpecifier, visitor: &mut dyn Visitor) {
+    if visitor.enter_storage_layout_specifier(target) {
+        accept_expression(&target.expression, visitor);
+    }
+    visitor.leave_storage_layout_specifier(target);
+}
+
+pub fn accept_interface_definition(target: &InterfaceDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_interface_definition(target) {
+        if let Some(ref inheritance) = target.inheritance {
+            accept_inheritance_specifier(inheritance, visitor);
+        }
+        accept_interface_members(&target.members, visitor);
+    }
+    visitor.leave_interface_definition(target);
+}
+
+pub fn accept_library_definition(target: &LibraryDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_library_definition(target) {
+        accept_library_members(&target.members, visitor);
+    }
+    visitor.leave_library_definition(target);
+}
+
+pub fn accept_struct_definition(target: &StructDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_struct_definition(target) {
+        accept_struct_members(&target.members, visitor);
+    }
+    visitor.leave_struct_definition(target);
+}
+
+pub fn accept_struct_member(target: &StructMember, visitor: &mut dyn Visitor) {
+    if visitor.enter_struct_member(target) {
+        accept_type_name(&target.type_name, visitor);
+    }
+    visitor.leave_struct_member(target);
+}
+
+pub fn accept_enum_definition(target: &EnumDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_enum_definition(target) {
+        accept_enum_members(&target.members, visitor);
+    }
+    visitor.leave_enum_definition(target);
+}
+
+pub fn accept_constant_definition(target: &ConstantDefinition, visitor: &mut dyn Visitor) {
```

Comment:
Is there a reason these are using dynamic dispatch instead of `&mut impl Visitor`?

Reply:
TBH, I wasn't paying attention to the difference between the two. I'll change it to `impl Visitor` because dynamic dispatch is slower, but OTOH it will almost certainly increase binary size as we expect to define multiple passes over the same IR.